sdk/storage/azure-storage-blob/tests/test_largest_block_blob.py

```diff
@@ -0,0 +1,348 @@
+# coding: utf-8
+
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+import pytest
+
+from os import path, remove, sys, urandom
+import platform
+import uuid
+
+from azure.core.pipeline.policies import HTTPPolicy
+from azure.storage.blob import (
+    BlobServiceClient,
+    BlobBlock
+)
+from azure.storage.blob._shared.base_client import format_shared_key_credential
+
+from _shared.testcase import StorageTestCase, GlobalStorageAccountPreparer
+
+# ------------------------------------------------------------------------------
+TEST_BLOB_PREFIX = 'largestblob'
+LARGEST_BLOCK_SIZE = 4000 * 1024 * 1024
+LARGEST_SINGLE_UPLOAD_SIZE = 5000 * 1024 * 1024
+
+# ------------------------------------------------------------------------------
+if platform.python_implementation() == 'PyPy':
+    pytest.skip("Skip tests for Pypy", allow_module_level=True)
+
+class StorageLargestBlockBlobTest(StorageTestCase):
+    def _setup(self, storage_account, key, additional_policies=None, min_large_block_upload_threshold=1 * 1024 * 1024,
+               max_single_put_size=32 * 1024):
+        self.bsc = BlobServiceClient(
+            self.account_url(storage_account, "blob"),
+            credential=key,
+            max_single_put_size=max_single_put_size,
+            max_block_size=LARGEST_BLOCK_SIZE,
+            min_large_block_upload_threshold=min_large_block_upload_threshold,
+            _additional_pipeline_policies=additional_policies)
+        self.config = self.bsc._config
+        self.container_name = self.get_resource_name('utcontainer')
+        self.container_name = self.container_name + str(uuid.uuid4())
+
+        if self.is_live:
+            self.bsc.create_container(self.container_name)
+
+    def _teardown(self, file_name):
+        if path.isfile(file_name):
+            try:
+                remove(file_name)
+            except:
+                pass
+
+    # --Helpers-----------------------------------------------------------------
+    def _get_blob_reference(self):
+        return self.get_resource_name(TEST_BLOB_PREFIX)
+
+    def _create_blob(self):
+        blob_name = self._get_blob_reference()
+        blob = self.bsc.get_blob_client(self.container_name, blob_name)
+        blob.upload_blob(b'')
+        return blob
+
+    # --Test cases for block blobs --------------------------------------------
+    @pytest.mark.live_test_only
+    @pytest.mark.skip(reason="This takes really long time")
```

Comment:
So this test will never run in CI right?

Reply:
Correct.
Some of them I marked as "live" so we want them in nightly builds - they consume quite a lot of memory/resources.
Some I explicitly skipped as they push real 4GB of data over network. Most extreme case is over 1hr due to https://github.com/Azure/azure-sdk-for-python/issues/11044 . So we'd keep them silent until we have better place for it.